src/components/Filters/SchemaSieve.ts

```diff
@@ -112,303 +168,106 @@ const generateFullTextSearchAjvFilter = (
-			// been flattened
-			if (!startsWith(field, delimiter)) {
-				result.properties![field] = value;
-				return result;
-			}
-
-			const keys = trimStart(field, delimiter).split(delimiter);
-			const lastKey = keys.pop()!;
-
-			let head = result;
-
-			// Convert keys array into nested object schemas
-			for (const key of keys) {
-				// Don't overwrite an existing entry
-				if (!head.properties![key]) {
-					head.properties![key] = {
-						type: 'object',
-						properties: {},
-					};
-				}
-
-				// Move the head pointer to the subschema that was just created
-				head = head.properties![key] as JSONSchema;
-			}
-
-			// Finally, use the last key to set the actual value of the field
-			head.properties![lastKey] = value;
-
-			return result;
-		},
-		base,
+	return Object.entries(model.operators(subSchema)).map(
+		([key, value]: [OperatorSlugs, string]) => ({
+			slug: key,
+			label: value,
+		}),
 	);
-
-	// If the schema uses `anyOf` then use recursion to populate the array
-	if (unflattenedSchema.anyOf) {
-		unflattenedSchema.anyOf = unflattenedSchema.anyOf.map((subSchema) =>
-			unflattenSchema(subSchema as JSONSchema, delimiter),
-		);
-	}
-
-	// Unflatten any required fields
-	if (unflattenedSchema.required) {
-		const requiredFields = unflattenedSchema.required;
-
-		// Reset the `required` array so it can be repopulated
-		unflattenedSchema.required = [];
-
-		requiredFields.forEach((requiredField) => {
-			const fields = trimStart(requiredField, delimiter).split(delimiter);
-
-			let head = unflattenedSchema;
-
-			for (const field of fields) {
-				if (!head.required) {
-					head.required = [];
-				}
-
-				if (!includes(head.required, field)) {
-					head.required.push(field);
-				}
-
-				head = head.properties![field] as JSONSchema;
-			}
-		});
-	}
-
-	return unflattenedSchema;
 };
 
-export const getCleanEditModel = (
+export const getModelFilter = (
 	schema: JSONSchema,
-	field?: string | null,
-) => {
-	if (!field) {
-		field = Object.keys(schema.properties!).shift()!;
-	}
-
-	const fieldOperators = getOperators(schema, field);
-	if (!fieldOperators.length) {
-		return {
-			field,
-			operator: '',
-			value: '',
-		};
-	}
-
-	const operator = fieldOperators.shift()!.slug;
-
-	let value: any = '';
-	const subschema = schema.properties![field];
-	if (typeof subschema !== 'boolean') {
-		if (subschema.enum) {
-			value = subschema.enum[0] || '';
-		}
-
-		if (subschema.oneOf) {
-			value = (subschema.oneOf[0] as JSONSchema).const || '';
-		}
-
-		if (subschema.type === 'boolean') {
-			value = true;
-		}
+	field: string,
+	operator: Operator<OperatorSlugs>,
+	value: any,
+	isFullTextSearch?: boolean,
+	operatorConverter?: (
+		operator: Operator<OperatorSlugs>,
+		operators: any,
+	) => Operator<OperatorSlugs>,
+): JSONSchema => {
+	let newOperator = operator;
+	const subSchema = isFullTextSearch
+		? { ...schema, oneOf: undefined, enum: undefined }
+		: schema;
```

Comment:
Should we try to keep the object a bit cleaner of props?
```suggestion
	let subSchema = schema;
	if (isFullTextSearch) {
		subSchema = {...schema};
		delete subSchema.oneOf;
		delete subSchema.enum;
	}
```